gotham/src/router/builder/single.rs

```diff
@@ -166,6 +167,97 @@ pub trait DefineSingleRoute {
     where
         NH: NewHandler + 'static;
 
+    /// Directs the route to serve static files from the given root path.
+    /// The route must contain a trailing glob segment, which will be used
+    /// to serve any matching names under the given path.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # extern crate gotham;
+    /// # extern crate hyper;
+    /// #
+    /// # use hyper::StatusCode;
+    /// # use gotham::router::Router;
+    /// # use gotham::router::builder::*;
+    /// # use gotham::pipeline::new_pipeline;
+    /// # use gotham::pipeline::single::*;
+    /// # use gotham::middleware::session::NewSessionMiddleware;
+    /// # use gotham::test::TestServer;
+    /// #
+    /// #
+    /// # fn router() -> Router {
+    /// #   let (chain, pipelines) = single_pipeline(
+    /// #       new_pipeline().add(NewSessionMiddleware::default()).build()
+    /// #   );
+    ///
+    /// build_router(chain, pipelines, |route| {
+    ///     route.get("/*").to_filesystem("resources/test/static_files");
+    /// })
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #   let test_server = TestServer::new(router()).unwrap();
+    /// #   let response = test_server.client()
+    /// #       .get("https://example.com/doc.html")
+    /// #       .perform()
+    /// #       .unwrap();
+    /// #   assert_eq!(response.status(), StatusCode::Ok);
+    /// # }
+    /// ```
+    fn to_filesystem(self, root: &str)
```

Comment:
I think `root` should be `AsRef<Path>` instead, right?

Reply:
If I'm understanding this suggestion correctly, this means we could use all sorts of `Path`, `OsStr`, `String`, `&str`, etc. values here instead of being restricted to `&str` or an automatically coerced `&String`. The practical benefit is that the path could be pulled from elsewhere (e.g. a config struct in a `lazy_static!`) and used directly rather than having to convert it for the `Router`API.

Is that accurate?

Reply:
`AsRef<Path>` has the benefit of allowing you to use anything that can be coerced into `Path`. So yes, that includes `String`, `&str`, `Path`, `PathBuf`, etc.

All types in the rust stdlib that deal with paths use it, for example [`std::fs::create_dir`](https://doc.rust-lang.org/std/fs/fn.create_dir.html)

Reply:
Thanks for confirming. Good idea.

Reply:
Great suggestion thanks - I'll take a crack at that change. I learn something new everyday :smile: .